src/Appwrite/Utopia/Request/Filters/V12.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Appwrite\Utopia\Request\Filters;
+
+use Appwrite\Utopia\Request\Filter;
+
+class V12 extends Filter
+{
+    // Convert 0.11 params format to 0.12 format
+    public function parse(array $content, string $model): array
+    {
+        switch ($model) {
+            // No IDs -> Custom IDs
+            case "account.create":
+            case "account.createMagicURLSession":
+            case "users.create":
+                $content = $this->addId($content, 'userId');
+                break;
+            case "functions.create":
+                $content = $this->addId($content, 'functionId');
+                break;
+            case "teams.create":
+                $content = $this->addId($content, 'teamId');
+                break;
+
+            // Status integer -> boolean
+            case "users.updateStatus":
+                $content = $this->convertStatus($content);
+                break;
+
+            // Deprecating order type
+            case "functions.listExecutions":
+                $content = $this->removeOrderType($content);
+                break;
+
+            // The rest (more complex) formats
+            case "database.createDocument":
+                $content = $this->addId($content, 'documentId');
+                $content = $this->removeParentProperties($content);
+                break;
+            case "database.listDocuments":
+                $content = $this->removeOrderCast($content);
+                $content = $this->convertOrder($content);
+                $content = $this->convertQueries($content);
+                break;
+            case "database.createCollection":
+                $content = $this->addId($content, 'collectionId');
+                $content = $this->removeRules($content);
+                $content = $this->addCollectionPermissionLevel($content);
+                break;
+            case "database.updateCollection":
+                $content = $this->removeRules($content);
+                $content = $this->addCollectionPermissionLevel($content);
+                break;
+        }
+
+        return $content;
+    }
+
+    // New parameters
+
+    protected function addId(array $content, string $key): array
+    {
+        $content[$key] = 'unique()';
+        return $content;
+    }
+
+    protected function addCollectionPermissionLevel(array $content): array
+    {
+        $content['permission'] = 'document';
```

Comment:
In `addCollectionPermissionLevel` why are we adding `document`? Shouldn't we be adding `collection `

Reply:
It' document-level. Migration tool does the same. We don't set it and that makes it document by default.